deploy/docker/docker-entrypoint.sh

```diff
@@ -110,13 +110,33 @@ CONFIG="${_EMQX_HOME}/etc/emqx.conf"
 CONFIG_PLUGINS="${_EMQX_HOME}/etc/plugins"
 for VAR in $(env)
 do
+    # echo value of $VAR_FULL_NAME
+    echo_value () { 
+        # check if contains sensitive value
+        for filter_keyword in 'PASSWORD' 'KEY' 'TOKEN' 'SECRET';do
+        if [ -z "${VAR_FULL_NAME##*$filter_keyword*}" ] ;then
+          IS_SENSITIVE=1
+          break
+        else
+          IS_SENSITIVE=0
+        fi
+        done
+
+        # echo value
+        if [ $IS_SENSITIVE == 1 ]
+        then
+            echo " ***secret***"
+        else
+            eval  echo \$$(echo $VAR_FULL_NAME)
+        fi
+    }
     # Config normal keys such like node.name = emqx@127.0.0.1
     if [[ ! -z "$(echo $VAR | grep -E '^EMQX_')" ]]; then
         VAR_NAME=$(echo "$VAR" | sed -r "s/EMQX_([^=]*)=.*/\1/g" | tr '[:upper:]' '[:lower:]' | sed -r "s/__/\./g")
         VAR_FULL_NAME=$(echo "$VAR" | sed -r "s/([^=]*)=.*/\1/g")
         # Config in emq.conf
         if [[ ! -z "$(cat $CONFIG |grep -E "^(^|^#*|^#*\s*)$VAR_NAME")" ]]; then
-            echo "$VAR_NAME=$(eval echo \$$VAR_FULL_NAME)"
+            echo "$VAR_NAME=$(echo_value)"
```

Comment:
`echo_value`

Reply:
resolved in 52c02a71dfc7f3331a1a80f5fcb52b9c42f410cc